base_layer/core/src/base_node/state_machine_service/state_machine.rs

```diff
@@ -100,26 +106,28 @@ impl<B: BlockchainBackend + 'static> BaseNodeStateMachine<B> {
         metadata_event_stream: Subscriber<ChainMetadataEvent>,
         config: BaseNodeStateMachineConfig,
         sync_validators: SyncValidators,
-        shutdown_signal: ShutdownSignal,
+        interrupt_signal: ShutdownSignal,
+        shutdown_trigger: Shutdown,
```

Comment:
What is the difference between `interrupt_signal` and `shutdown_trigger`. They sound really similar

Reply:
The interrupt signal is an external signal that tells the state machine it should shutdown. The Shutdown Signal is how the state machine indicates to the Application that it has shutdown. Suggestions for a clearer name?

Reply:
Perhaps `should_shutdown` and `on_shutdown`?

Reply:
changed it to `service_shutdown`.